python/pyspark/sql/pandas/utils.py

```diff
@@ -84,3 +87,57 @@ def pyarrow_version_less_than_minimum(minimum_pyarrow_version: str) -> bool:
         return False
 
     return LooseVersion(pyarrow.__version__) < LooseVersion(minimum_pyarrow_version)
+
+
+def barrier(f: Callable) -> Callable:
```

Comment:
If we're going to have this standalone API, this should work together with other similar API like `groupby().applyInPandas`.